File: Itido-BMS/dart/bms_dart/lib/src/models/work_replacement.dart
import 'package:bms_dart/src/models/absence.dart';
import 'package:bms_dart/src/models/contract.dart';

import 'user.dart';

class WorkReplacement {
  // int workId;
  // int contractId;
  // int absenceId;

  Absence absence;
  Contract contract;

  WorkReplacement();

  factory WorkReplacement.fromJson(json, {rootJson, path}) {
    if (json == null) return null;
    return WorkReplacement._fromJson(json, rootJson ?? {}, path ?? '');
  }

  WorkReplacement._fromJson(json, rootJson, path)
      :
        // this.workId = json['workId'],
        // this.contractId = json['contractId'],
        // this.absenceId = json['absenceId'],
        this.absence = Absence.fromJson(
            json['absence'] ?? rootJson['${path}_absence'],
            rootJson: rootJson,
            path: '${path}_absence'),
        this.contract = Contract.fromJson(
            json['contract'] ?? rootJson['${path}_contract'],
            rootJson: rootJson,
            path: '${path}_contract');

  // Map<String, dynamic> toMap() => {
  //       'workId': this.workId,
  //       'contractId': this.contractId,
  //       'absenceId': this.absenceId,
  //     };
}

File: Itido-BMS/RentAPI/Rent/Models/TimePlanning/WorkReplacement.cs
using System;
using System.Linq.Expressions;
using Rent.Helpers;

namespace Rent.Models.TimePlanning
{
    public class WorkReplacement
    {
        public int ID { get; set; }
        public int WorkID { get; set; }
        public int? ContractID { get; set; }
        public int AbsenceID { get; set; }
        public virtual Contract Contract { get; set; }
        public virtual Absence Absence { get; set; }
        public virtual Work Work { get; set; }


        public static Expression<Func<WorkReplacement, dynamic>> DetailedDTO()
        {
            return v => v != null ?
            new
            {
                v.ID,
                //Absence = Absence.BasicDTO(requester, requesterRole).Compile()(v.Absence),
                //AbsenceReason = v.Absence != null ? v.Absence.AbsenceReason : null,
                //Contract = Contract.BasicDTO().CallWithArg(v.Contract),

            } : null;
        }

        public static Expression<Func<WorkReplacement, dynamic>> BasicDTO()
        {
            return v => v != null ?
            new
            {
                v.ID,
                //AbsenceReason = v.Absence != null ? v.Absence.AbsenceReason : null,
                //Contract = Contract.BasicDTO().CallWithArg(v.Contract),
            } : null;
        }
    }
}